### PR TITLE
🤡 Auth & Discussions fixes after new global nav release 🤡

### PR DESCRIPTION
**Pull request by nikodamn**
(mostly) Auth & Discussions fixes after new global nav release 

Additionally :
* Rename `TopBar` to `GlobalNavigationMobile` and it's refactoring 🔧 
* Remove `Navigation`, since there's no side-menu anymore on Mobile. It's functionality was incorporated into `GlobalNavigation` and `GlobalNavigationMobile` at least until new local nav is implemented ;) 🔥 
* Slight refactoring of the Auth/Login parts 🔨 
* Move and rename `MercuryNavigationTests` to Discussions page, since it is testing the Discussion's app navigation 😨 

@ludwikkazmierczak @sczerwinski-wikia @chessky @qaga

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SearchPageObject.java#L20](https://github.com/Wikia/selenium-tests/blob/ed73fc12ae24a305861afbc5831bb8f8ff6ed05a/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/SearchPageObject.java#L20): Remove this empty statement. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)
1. ![MINOR][MINOR] [EmailUtils.java#L26](https://github.com/Wikia/selenium-tests/blob/ed73fc12ae24a305861afbc5831bb8f8ff6ed05a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L26): Move this constructor to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)
1. ![MINOR][MINOR] [EmailUtils.java#L35](https://github.com/Wikia/selenium-tests/blob/ed73fc12ae24a305861afbc5831bb8f8ff6ed05a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L35): Define a constant instead of duplicating this literal "imaps" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [EmailUtils.java#L54](https://github.com/Wikia/selenium-tests/blob/ed73fc12ae24a305861afbc5831bb8f8ff6ed05a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L54): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS134)
1. ![MINOR][MINOR] [EmailUtils.java#L99](https://github.com/Wikia/selenium-tests/blob/ed73fc12ae24a305861afbc5831bb8f8ff6ed05a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L99): Move the array designator from the variable to the type. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1197)
1. ![MINOR][MINOR] [EmailUtils.java#L122](https://github.com/Wikia/selenium-tests/blob/ed73fc12ae24a305861afbc5831bb8f8ff6ed05a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L122): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SearchPageObject.java#L20](https://github.com/Wikia/selenium-tests/blob/9a4134880046e73acd2351b744200e641a63e608/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/SearchPageObject.java#L20): Remove this empty statement. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)
1. ![MINOR][MINOR] [EmailUtils.java#L26](https://github.com/Wikia/selenium-tests/blob/9a4134880046e73acd2351b744200e641a63e608/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L26): Move this constructor to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)
1. ![MINOR][MINOR] [EmailUtils.java#L35](https://github.com/Wikia/selenium-tests/blob/9a4134880046e73acd2351b744200e641a63e608/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L35): Define a constant instead of duplicating this literal "imaps" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [EmailUtils.java#L54](https://github.com/Wikia/selenium-tests/blob/9a4134880046e73acd2351b744200e641a63e608/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L54): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS134)
1. ![MINOR][MINOR] [EmailUtils.java#L99](https://github.com/Wikia/selenium-tests/blob/9a4134880046e73acd2351b744200e641a63e608/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L99): Move the array designator from the variable to the type. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1197)
1. ![MINOR][MINOR] [EmailUtils.java#L122](https://github.com/Wikia/selenium-tests/blob/9a4134880046e73acd2351b744200e641a63e608/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L122): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/testcases/discussions/NavigationTests.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import com.wikia.webdriver.elements.mercury.pages.discussions.GuidelinesPage;
+import org.testng.annotations.Test;
+
+@Test(groups = "Mercury_Navigation")
+@InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+@Execute(onWikia = MercuryWikis.DISCUSSIONS_MOBILE)
+public class NavigationTests {
+
+  @Test(groups = "mobileWiki_navigation_openAndCloseNavigationAndItsSubMenu")
+  public void mercury_navigation_openAndCloseNavigationAndItsSubMenu() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/discussions/NavigationTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/discussions/NavigationTests.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    GlobalNavigationMobile nav = new GuidelinesPage().open().getGlobalNavigationMobile().openNavigation();
+    Assertion.assertTrue(nav.isFirstLevelMenuVisible());
+
+    nav.clickCloseButton();
+    Assertion.assertFalse(nav.isFirstLevelMenuVisible());
+  }
+
+  @Execute(onWikia = MercuryWikis.DE_WIKI, language = "de")
+  @Test(groups = "mobileWiki_navigation_navigationOnNonEnglishWiki")
+  public void mercury_navigation_navigationOnNonEnglishWiki() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/discussions/NavigationTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mobilewikitests/NavigationMobileWikiTests.java [hunk 3]
```diff
@@ -2,40 +2,47 @@
...
 
 @Test(groups = "MobileWiki_Navigation")
 @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
 @Execute(onWikia = MercuryWikis.MERCURY_AUTOMATION_TESTING)
-public class NavigationMobileWikiTests extends NavigationTests {
+public class NavigationMobileWikiTests {
 
   //test disabled until we find solution how to click 'Wikis' link in navigation
   @Test(groups = "mobileWiki_navigation_openAndCloseNavigationAndItsSubMenu", enabled = false)
   public void mobileWiki_navigation_openAndCloseNavigationAndItsSubMenu() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mobilewikitests/NavigationMobileWikiTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mobilewikitests/NavigationMobileWikiTests.java [hunk 4]
```diff
@@ -2,40 +2,47 @@
...
+    GlobalNavigationMobile nav = new ArticlePage().open(MercurySubpages.MAIN_PAGE).getGlobalNavigationMobile().openNavigation();
+    Assertion.assertTrue(nav.isFirstLevelMenuVisible());
+
+    nav.clickCloseButton();
+    Assertion.assertFalse(nav.isFirstLevelMenuVisible());
   }
 
   @Execute(onWikia = MercuryWikis.DE_WIKI, language = "de")
   @Test(groups = "mobileWiki_navigation_navigationOnNonEnglishWiki")
   public void mobileWiki_navigation_navigationOnNonEnglishWiki() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mobilewikitests/NavigationMobileWikiTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mobilewikitests/NavigationMobileWikiTests.java [hunk 5]
```diff
@@ -2,40 +2,47 @@
...
-    );
+    GlobalNavigationMobile nav = new ArticlePage().open(MercurySubpages.MAIN_PAGE).getGlobalNavigationMobile().openNavigation();
+    Assertion.assertTrue(nav.isFirstLevelMenuVisible());
+
+    nav.clickCloseButton();
+    Assertion.assertFalse(nav.isFirstLevelMenuVisible());
   }
 
   @Test(groups = "mobileWiki_navigation_navigationOnEnglishWiki")
   public void mobileWiki_navigation_navigationOnEnglishWiki() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mobilewikitests/NavigationMobileWikiTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mobilewikitests/ArticlePageTests.java [hunk 6]
```diff
@@ -133,28 +130,28 @@ public void mercury_article_navigateToArticlesWithColonAndQuestionMark() {
 
     Log.warning("Info", "Accessing article through link in navigation side");
 
-    topBar.openNavigation();
+    globalNavigationMobile.openNavigation();
     //navigation.openSubMenu(3);
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mobilewikitests/ArticlePageTests.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: src/test/java/com/wikia/webdriver/testcases/mobilewikitests/ArticlePageTests.java [hunk 7]
```diff
@@ -133,28 +130,28 @@ public void mercury_article_navigateToArticlesWithColonAndQuestionMark() {
...
-    navigation.openPageLink(5);
+    //navigation.openPageLink(5);
 
     Assertion.assertFalse(driver.getCurrentUrl().contains(encodedColonUrl));
     Assertion.assertTrue(MercurySubpages.COLON.toLowerCase().contains(article.getArticleTitle().toLowerCase()));
 
-    topBar.openNavigation();
+    globalNavigationMobile.openNavigation();
     //navigation.openSubMenu(3);
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mobilewikitests/ArticlePageTests.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 13 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SearchPageObject.java#L20](https://github.com/Wikia/selenium-tests/blob/90c08fc0f4e81424dc86939cc53fdd92730d5b2c/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/SearchPageObject.java#L20): Remove this empty statement. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)
1. ![MINOR][MINOR] [EmailUtils.java#L26](https://github.com/Wikia/selenium-tests/blob/90c08fc0f4e81424dc86939cc53fdd92730d5b2c/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L26): Move this constructor to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)
1. ![MINOR][MINOR] [EmailUtils.java#L35](https://github.com/Wikia/selenium-tests/blob/90c08fc0f4e81424dc86939cc53fdd92730d5b2c/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L35): Define a constant instead of duplicating this literal "imaps" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [EmailUtils.java#L54](https://github.com/Wikia/selenium-tests/blob/90c08fc0f4e81424dc86939cc53fdd92730d5b2c/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L54): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS134)
1. ![MINOR][MINOR] [EmailUtils.java#L99](https://github.com/Wikia/selenium-tests/blob/90c08fc0f4e81424dc86939cc53fdd92730d5b2c/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L99): Move the array designator from the variable to the type. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1197)
1. ![MINOR][MINOR] [EmailUtils.java#L122](https://github.com/Wikia/selenium-tests/blob/90c08fc0f4e81424dc86939cc53fdd92730d5b2c/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L122): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/common/core/EmailUtils.java [hunk 8]
```diff
@@ -64,7 +57,7 @@ public static String getFirstEmailContent(String userName, String password, Stri
           }
         }
         
-        if (i > 15) {
+        if (i > 30) {
```
Comment (nikodamn) [inline src/test/java/com/wikia/webdriver/common/core/EmailUtils.java]: dirty

**Comment by WikiaTech** [comment]
SonarQube analysis reported 12 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [EmailUtils.java#L26](https://github.com/Wikia/selenium-tests/blob/c05d0418fabadd8a10b29f24dbda90382a3b115a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L26): Move this constructor to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)
1. ![MINOR][MINOR] [EmailUtils.java#L35](https://github.com/Wikia/selenium-tests/blob/c05d0418fabadd8a10b29f24dbda90382a3b115a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L35): Define a constant instead of duplicating this literal "imaps" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [EmailUtils.java#L54](https://github.com/Wikia/selenium-tests/blob/c05d0418fabadd8a10b29f24dbda90382a3b115a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L54): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS134)
1. ![MINOR][MINOR] [EmailUtils.java#L99](https://github.com/Wikia/selenium-tests/blob/c05d0418fabadd8a10b29f24dbda90382a3b115a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L99): Move the array designator from the variable to the type. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1197)
1. ![MINOR][MINOR] [EmailUtils.java#L122](https://github.com/Wikia/selenium-tests/blob/c05d0418fabadd8a10b29f24dbda90382a3b115a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L122): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [EmailUtils.java#L26](https://github.com/Wikia/selenium-tests/blob/e4c99a64fbb8eda60cbc5295d2b782def56f271a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L26): Move this constructor to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)
1. ![MINOR][MINOR] [EmailUtils.java#L35](https://github.com/Wikia/selenium-tests/blob/e4c99a64fbb8eda60cbc5295d2b782def56f271a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L35): Define a constant instead of duplicating this literal "imaps" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [EmailUtils.java#L54](https://github.com/Wikia/selenium-tests/blob/e4c99a64fbb8eda60cbc5295d2b782def56f271a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L54): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS134)
1. ![MINOR][MINOR] [EmailUtils.java#L99](https://github.com/Wikia/selenium-tests/blob/e4c99a64fbb8eda60cbc5295d2b782def56f271a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L99): Move the array designator from the variable to the type. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1197)
1. ![MINOR][MINOR] [EmailUtils.java#L122](https://github.com/Wikia/selenium-tests/blob/e4c99a64fbb8eda60cbc5295d2b782def56f271a/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L122): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/testcases/mobilewikitests/ArticlePageTests.java (hunk 7 above)
Comment (nikodamn) [reply src/test/java/com/wikia/webdriver/testcases/mobilewikitests/ArticlePageTests.java]: This test is disabled and it's not under my jurisdiction. 

gandalf

File: src/test/java/com/wikia/webdriver/testcases/mobilewikitests/ArticlePageTests.java (hunk 6 above)
Comment (nikodamn) [reply src/test/java/com/wikia/webdriver/testcases/mobilewikitests/ArticlePageTests.java]: This test is disabled and it's not under my jurisdiction. 

gandalf

File: src/test/java/com/wikia/webdriver/common/core/EmailUtils.java (hunk 8 above)
Comment (mkjmdski) [reply src/test/java/com/wikia/webdriver/common/core/EmailUtils.java]: what's an 'i'? Try to use more explicit variable names.

Comment (nikodamn) [reply src/test/java/com/wikia/webdriver/common/core/EmailUtils.java]: Good point, I just updated the param value... But yeah, scout rule. ;)

Comment (nikodamn) [reply src/test/java/com/wikia/webdriver/common/core/EmailUtils.java]: I have to admit that question what's an "I" is one of the most magnificent puzzles of all time. Can't give you a shot answer, but maybe on of multiple philosophers of ages, would be able to satisfy you with an answer. :trollface:

**Pull request merged**